build-logic/src/main/kotlin/io/openfeedback/extensions/ProjectExt.kt

```diff
@@ -38,28 +44,57 @@ fun Project.publishIfNeededTaskProvider(): TaskProvider<Task> {
     }
 }
 
-fun Project.getOssStagingUrl(): String {
-    val url = try {
-        this.extensions.extraProperties["ossStagingUrl"] as String?
+private val baseUrl = "https://s01.oss.sonatype.org/service/local/"
+
+private val nexusStagingClient by lazy {
+    NexusStagingClient(
+        baseUrl = baseUrl,
+        username = System.getenv(EnvVarKeys.Nexus.username) ?: error("please set the ${EnvVarKeys.Nexus.username} environment variable"),
+        password = System.getenv(EnvVarKeys.Nexus.password) ?: error("please set the ${EnvVarKeys.Nexus.password} environment variable"),
+    )
+}
+
+
+internal fun Project.getOssStagingRepoId(): String? {
+    return try {
+        this.extensions.extraProperties["ossStagingRepositoryId"] as String?
     } catch (ignored: ExtraPropertiesExtension.UnknownPropertyException) {
         null
     }
-    if (url != null) {
-        return url
+}
+
+private fun Project.getOrCreateOssStagingRepoId(): String {
```

Comment:
If X modules are executed in parallel and they check the nullability of `repoId` at the same time, what is the impact of `nexusStagingClient.createRepository` X times?

Reply:
> what is the impact of nexusStagingClient.createRepository X times?

It's bad! I'll add a lock

Reply:
done